### PR TITLE
R2r cat

### PR DESCRIPTION
**Pull request by radare**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: binr/r2r/r2r.c [hunk 1]
```diff
@@ -158,6 +168,9 @@ int main(int argc, char **argv) {
 		case 'C':
 			r2r_dir = opt.arg;
 			break;
+		case 'n':
```
Comment (ret2libc) [inline binr/r2r/r2r.c]: this is not in the help. What does it do?

File: binr/r2r/r2r.c [hunk 2]
```diff
@@ -82,6 +84,10 @@ static bool r2r_chdir(const char *argv0) {
 #endif
 }
 
+static void r2r_test_run_unit() {
```
Comment (ret2libc) [inline binr/r2r/r2r.c]: `static void r2r_test_run_unit(void)`

File: binr/r2r/r2r.c (hunk 1 above)
Comment (thestr4ng3r) [reply binr/r2r/r2r.c]: It does nothing

Comment (ret2libc) [reply binr/r2r/r2r.c]: was that serious? :D

Comment (thestr4ng3r) [reply binr/r2r/r2r.c]: Yes, see line 267.

File: binr/r2r/r2r.c [hunk 3]
```diff
@@ -236,6 +270,11 @@ int main(int argc, char **argv) {
 	}
 
 	R_FREE (cwd);
+	uint32_t loaded_tests = r_pvector_len (&state.db->tests);
+	printf ("Loaded %u tests.\n", loaded_tests);
+	if (nothing) {
+		return 0;
```
Comment (thestr4ng3r) [inline binr/r2r/r2r.c]: This will leak everything.

File: binr/r2r/r2r.c (hunk 1 above)
Comment (ret2libc) [reply binr/r2r/r2r.c]: mh, ok... it should be listed in the help msg anyway though

Comment (radare) [reply binr/r2r/r2r.c]: fixed

File: binr/r2r/r2r.c (hunk 2 above)
Comment (radare) [reply binr/r2r/r2r.c]: i fixed this in 10 other places inside run.c, thanks for pointing out

File: binr/r2r/r2r.c (hunk 3 above)
Comment (radare) [reply binr/r2r/r2r.c]: fixing now

File: binr/r2r/r2r.c [hunk 4]
```diff
@@ -34,18 +34,21 @@ static void print_state(R2RState *state, ut64 prev_completed);
 static void print_log(R2RState *state, ut64 prev_completed, ut64 prev_paths_completed);
 
 static int help(bool verbose) {
-	printf ("Usage: r2r [-vh] [-j threads] [test path]\n");
+	printf ("Usage: r2r [-vh] [-j threads] [test file/dir | .test-type]\n");
```
Comment (ret2libc) [inline binr/r2r/r2r.c]: It seems like the special char is `@` and not `.`

File: binr/r2r/r2r.c [hunk 5]
```diff
@@ -217,7 +232,28 @@ int main(int argc, char **argv) {
...
+				arg++;
+				eprintf ("Category: %s\n", arg);
+				if (!strcmp (arg, "unit")) {
+					r2r_test_run_unit ();
+					continue;
+				} else if (!strcmp (arg, "fuzz")) {
+					eprintf (".fuzz: TODO\n");
+					continue;
+				} else if (!strcmp (arg, "json")) {
+					arg = "../../test/new/db/json";
```
Comment (ret2libc) [inline binr/r2r/r2r.c]: so @thestr4ng3r @radare this assume that r2r is always run from a particular dir, right? :(

Comment (radare) [reply binr/r2r/r2r.c]: yes, the database directory and i can probably remove all the `../../test/new/db` in here..

Comment (radare) [reply binr/r2r/r2r.c]: i'll rework this in a separate pr for a better proposal

File: binr/r2r/r2r.c (hunk 4 above)
Comment (radare) [reply binr/r2r/r2r.c]: fixed

File: binr/r2r/r2r.c (hunk 5 above)
Comment (radare) [reply binr/r2r/r2r.c]: nvm fixed in here

**Pull request merged**